testnet/stacks-node/src/config.rs

```diff
@@ -2546,6 +2551,13 @@ pub struct MinerConfigFile {
 
 impl MinerConfigFile {
     fn into_config_default(self, miner_default_config: MinerConfig) -> Result<MinerConfig, String> {
+        match &self.mining_key {
```

Comment:
As long as a node.seed is set up, this will never trigger. Should it be removed? Or do we prefer this and remove the mining-key being automatically set?

Reply:
What if `node.seed` isn't configured and the node doesn't have `miner.seed = true` - will this cause the node to panic?

Reply:
I've tested it using `stacks-regtest-env` for the config with 
1. `node.miner = false` and `node.seed` empty with miner config data in it
2. `node.miner = false` and `node.seed` empty without miner config in it specified. 
3. `node.miner = true` and `node.seed` specified
4. `node.miner = true` and `miner.mining_key` specified
On all these cases it didn't panic.

It only panicked when `node.miner = true` and `node.seed` and `miner.mining_key` were not provided.